5-VM-Ubuntu-1-NodeTypes-Secure/azuredeploy.json

```diff
@@ -83,155 +96,212 @@
+      "type": "string",
+      "allowedValues": [
+        "Standard_LRS",
+        "Standard_GRS"
+      ],
+      "defaultValue": "Standard_LRS",
       "metadata": {
-        "description": "Instance count for node type"
+        "description": "Replication option for the support log storage account"
       }
     },
-    "nodeDataDrive": {
+    "applicationDiagnosticsStorageAccountType": {
       "type": "string",
-      "defaultValue": "Temp",
       "allowedValues": [
-        "OS", "Temp"
+        "Standard_LRS",
+        "Standard_GRS"
       ],
+      "defaultValue": "Standard_LRS",
       "metadata": {
-        "description": "The drive to use to store data on a cluster node."
+        "description": "Replication option for the application diagnostics storage account"
       }
     },
-    "nodeTypeSize": {
+    "nt0InstanceCount": {
+      "type": "int",
+      "defaultValue": 5,
+      "metadata": {
+        "description": "Instance count for node type"
+      }
+    },
+    "vmNodeType0Size": {
       "type": "string",
       "defaultValue": "Standard_D2_v2",
       "metadata": {
-        "description": "The VM size to use for cluster nodes."
+        "description": "VM Type"
       }
     }
   },
   "variables": {
-    "vmNodeType0Name": "nodetype0-vm",
-    "vmNodeType0Size": "[parameters('nodeTypeSize')]",
+    "computeLocation": "[parameters('clusterLocation')]",
+    "dnsName": "[parameters('clusterName')]",
+    "vmStorageAccountName": "[toLower(concat( 'sflogs', uniqueString(resourceGroup().id), '1' ))]",
+    "vmName": "vm",
+    "publicIPAddressName": "PublicIP-VM",
+    "publicIPAddressType": "Dynamic",
+    "vmStorageAccountContainerName": "vhds",
     "virtualNetworkName": "VNet",
+    "addressPrefix": "10.0.0.0/16",
     "nicName": "NIC",
-    "lbIp0Name": "PublicIP-LB-FE-0",
+    "lbName": "LoadBalancer",
+    "lbIPName": "PublicIP-LB-FE",
+    "availSetName": "AvailabilitySet",
+    "maxPercentUpgradeDomainDeltaUnhealthyNodes": "100",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "overProvision": "false",
+    "vmssApiVersion": "2017-03-30",
+    "sfrpApiVersion": "2017-07-01-preview",
+    "lbApiVersion": "2015-06-15",
+    "vNetApiVersion": "2015-06-15",
+    "storageApiVersion": "2016-01-01",
+    "publicIPApiVersion": "2015-06-15",
+    "nt0applicationStartPort": "20000",
+    "nt0applicationEndPort": "30000",
+    "nt0ephemeralStartPort": "49152",
+    "nt0ephemeralEndPort": "65534",
     "nt0fabricTcpGatewayPort": "19000",
     "nt0fabricHttpGatewayPort": "19080",
     "subnet0Name": "Subnet-0",
-    "subnet0Ref": "[resourceId('Microsoft.Network/virtualNetworks/subnets/', variables('virtualNetworkName'), variables('subnet0Name'))]",
-    "logStorageAccountName": "[concat('log', uniqueString(resourceGroup().id))]",
+    "subnet0Prefix": "10.0.0.0/24",
+    "subnet0Ref": "[concat(variables('vnetID'),'/subnets/',variables('subnet0Name'))]",
```

Comment:
use resourceId

resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnet0Name'))

Reply:
Updated "subnet0Ref" value to use resourceId